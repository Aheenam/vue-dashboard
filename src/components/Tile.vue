<template>
  <div class="tile" :style="tile">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'tile',
  props: {
    x: {
      type: Number,
      required: true
    },
    y: {
      type: Number,
      required: true
    },
    w: {
      type: Number,
      required: true
    },
    h: {
      type: Number,
      required: true
    }
  },
  data () {
    return {
      tile: {
        gridColumnStart: this.x,
        gridColumnEnd: this.x + this.w,
        gridRowStart: this.y,
        gridRowEnd: this.y + this.h
      }
    }
  }
}
</script>

<style>
</style>
