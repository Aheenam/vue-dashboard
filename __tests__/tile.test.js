import Tile from '../src/Tile'
import Vue from 'vue'

describe('tile-test', () => {

    const Constructor = Vue.extend(Tile);
    const vm = new Constructor({
        propsData: { x: 1, y: 1, w: 1, h: 1 }
      }).$mount()

    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    });

})
