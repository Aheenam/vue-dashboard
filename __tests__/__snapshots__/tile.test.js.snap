// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tile-test should match the snapshot 1`] = `
<div
  class="tile"
/>
`;
